data-structures-csharp/data-structures-csharp/CircularBuffer/CircularBuffer.cs

```diff
@@ -6,19 +6,20 @@
 namespace DataStructures.CircularBufferSpace
 {
     [Serializable]
-    public class CircularBuffer<T> : IEnumerable<T>, ICollection<T>
```

Comment:
Oh, that has to do with it being a FIFO data type. The ICollection<T> interface (the generic version of ICollection) requires implementation of a method that allows you to remove any designatedd element of the buffer when you should only be able to remove the oldest element.